flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1759,11 +1759,22 @@ class JobManager(
       case None => None
     }
 
-    // remove all job-related BLOBs from local and HA store
-    libraryCacheManager.unregisterJob(jobID)
-    blobServer.cleanupJob(jobID, removeJobFromStateBackend)
+    // remove all job-related BLOBs from local and HA store, only if the job was removed correctly
+    futureOption match {
+      case Some(future) => future.onComplete{
+        case scala.util.Success(_) => {
+          libraryCacheManager.unregisterJob(jobID)
+          blobServer.cleanupJob(jobID, removeJobFromStateBackend)
+          jobManagerMetricGroup.removeJob(jobID)
```

Comment:
I think we could always execute this call independent of whether the removal from the `SubmittedJobGraphStore` was successful or not.